fs-api/fs-api.cabal

```diff
@@ -35,6 +35,8 @@ library
     System.FS.Condense
     System.FS.CRC
     System.FS.IO
+
+  other-modules:
     System.FS.IO.Internal
     System.FS.IO.Internal.Handle
```

Comment:
Weakly opposed to add stuff to `other-modules`, eg I could see unwrapping `HandleIO`/`FHandle` to their platform-specific internals be useful in certain corner cases (where people will happily accept to import internal modules and be ok that we don't guarantee any kind of stability for these). `other-modules` makes this much more annoying (having to modify the source code, or [dark arts](https://www.tweag.io/blog/2021-01-07-haskell-dark-arts-part-i/)).

An alternative would be to do sth like what [this blog post](https://nikita-volkov.github.io/internal-convention-is-a-mistake/) argues for, but given that the pattern advocated in there is very uncommon, internal modules are fine for now.

(Also see the [corresponding Reddit discussion](https://redd.it/a0ioo3) for more takes on this.)

Reply:
I've seen the blogpost before, and I decided on hiding the modules in part because of it. The blogpost advocates for not using the Internal convention, but then there are two options: putting the modules in `other-modules`, or properly considering the modules for versioning.

I do agree with the sentiment that the modules could be useful for downstream packages. I've thought a bit more about whether I should un-internalise these modules, and I decided to do so. This means the modules are now properly considered for the versioning, but I think that's okay. The modules are relatively stable, and the user can't really break much if we do. If it turns out that these modules become a maintenance burden in the end, then I'll move them to their own package, which seems to be in line with what the blogpost suggests